core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -272,18 +272,25 @@ object AdminUtils extends Logging with AdminUtilities {
                     numPartitions: Int = 1,
                     replicaAssignmentStr: String = "",
                     checkBrokerAvailable: Boolean = true,
-                    rackAwareMode: RackAwareMode = RackAwareMode.Enforced) {
+                    rackAwareMode: RackAwareMode = RackAwareMode.Enforced,
+                    validateOnly: Boolean = false) {
+    if (zkUtils.pathExists(ZkUtils.ReassignPartitionsPath)) {
+      // We prevent addition partitions while a reassignment is in progress, since
+      // during reassignment there is no meaningful notion of replication factor
+      throw new ReassignmentInProgressException("A partition reassignment is in progress.")
+    }
+
     val existingPartitionsReplicaList = zkUtils.getReplicaAssignmentForTopics(List(topic))
```

Comment:
Maybe rename this to the more concise `existingAssignment`. Similar for other unnecessarily long variable names.